docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+and elsewhere in the file system directory structure.
+
+Dependencies
+~~~~~~~~~~~~
+
+The dependencies of a part are described using the ``build-snaps`` and
+``build-packages`` definitions. These specify lists of snaps and system
+packages to be installed before the part is built.
+
+System packages are referred to by the names that they are identified by on
+the host system, and they are installed using the host's native package
+manager, such as :command:`apt` or :command:`dnf`.
+
+Build process
+~~~~~~~~~~~~~
+
+Every part needs to define how it will be built. Since all parts are built
+using *plugins*, each part definition will use the ``plugin`` key.
+
+The available plugins are provided as modules in the `craft_parts/plugins`
+directory -- see :py:mod:`craft_parts.plugins`. These aim to simplify the
+process of building source code written in a variety of programming languages
+using appropriate build systems, libraries and frameworks. If a plugin is not
+available for a particular combination of these attributes, a custom plugin
+can be created or a basic plugin can be used to manually specify the build
+actions to be taken, using the ``override-build`` key.
+
+When a plugin is used, it exposes additional keys that can be used to
+define behaviour that is specific to the type of project that the plugin
+supports. For example, the :py:mod:`cmake plugin <craft_parts.plugins.cmake_plugin>` provides the ``cmake-parameters`` and
+``cmake-generator`` keys that can be used to configure how :command:`cmake`
+is used in the build process.
+
+The build process can be further customised with the ``build-environment``,
+``build-attributes`` and ``override-build`` keys.
+
+The ``build-environment`` key defines assignments to shell environment variables in the build environment, specified as a list of key-value pairs.
+In YAML format, this can take either of the following forms:
+
+.. code:: yaml
+
+   build-environment: [MESSAGE: "Hello world", NAME: "Craft Parts"]
+
+   build-environment:
+     - MESSAGE: "Hello world"
+     - NAME: "Craft Parts"
+
+The ``build-attributes`` key allows a number of standard customisations to be
+applied to the build. Some of these are used to address issues that occur in
+specific situations; others, such as ``debug`` are generally useful.
+
+By default, Craft Parts builds independent parts in parallel. This can be
+disabled by setting the ``disable-parallel`` key to ``True``.
+
+The ``override-build`` key is used to override the build process provided by
+a plugin, and it can be used to replace or extend it.
+
+Build products
+~~~~~~~~~~~~~~
+
+After a part has been built in the *build* step, the build products are
+prepared in three further steps.
+
+If the ``organize`` key is specified, it is used to move files
+specify a list of mappings between files
+in the part key-value pairs
+
+   * How to manage the different steps in the build process
```

Comment:
Is this section is a todo list?

Reply:
No, it's just working that got caught up in the commit. Sorry for the confusion.